pkg/scanners/scannerv4/convert.go

```diff
@@ -52,22 +66,63 @@ func components(metadata *storage.ImageMetadata, report *v4.VulnerabilityReport)
 	return components
 }
 
-func layerIndex(layerSHAToIndex map[string]int32, report *v4.VulnerabilityReport, pkgID string) *storage.EmbeddedImageScanComponent_LayerIndex {
-	// TODO(ROX-21377): Confirm with Clair team how handle multiple environments
-	envList := report.GetContents().GetEnvironments()[pkgID]
-	if len(envList.GetEnvironments()) > 0 {
-		env := envList.GetEnvironments()[0]
+func environment(report *v4.VulnerabilityReport, id string) *v4.Environment {
+	envList, ok := report.GetContents().GetEnvironments()[id]
+	if !ok {
+		return nil
+	}
 
-		if val, ok := layerSHAToIndex[env.GetIntroducedIn()]; ok {
-			return &storage.EmbeddedImageScanComponent_LayerIndex{
-				LayerIndex: val,
-			}
-		}
+	envs := envList.GetEnvironments()
+	if len(envs) > 0 {
+		// Just use the first environment.
+		return envs[0]
 	}
 
 	return nil
 }
 
+func location(env *v4.Environment) string {
+	// Assume : is an invalid character for a filepath in a Linux filesystem.
```

Comment:
I'm not sure if this is a safe assumption. I can make `h:e:llo` on macOS, for example. Perhaps it's best to keep the location as a 1:1 mapping from PackageDB?

Reply:
Agreed, see comment https://github.com/stackrox/stackrox/pull/9444#discussion_r1459250632 - believe a better approach would be to strip the source if recognized, otherwise leave it.